ssl/statem/statem_clnt.c

```diff
@@ -3452,6 +3452,11 @@ MSG_PROCESS_RETURN tls_process_hello_req(SSL *s, PACKET *pkt)
         return MSG_PROCESS_ERROR;
     }
 
+    if ((SSL_get_options(s) & SSL_OP_NO_RENEGOTIATION)) {
+        ssl3_send_alert(s, SSL3_AL_WARNING, SSL_AD_NO_RENEGOTIATION);
+        return MSG_PROCESS_FINISHED_READING;
+    }
```

Comment:
This is during the processing of the HelloRequest. If nothing is done here, then the calls to SSL_renegotiate()/SSL_renegotiate_abbreviated() that follow error out, and unit-test 12-no-renegotiation-client-by-server has a result of ClientFail. I think that sending the alert is the right thing to do. I had to add the TimeoutError because the server never receives this alert, and I couldn't figure out why. I would prefer to get the alert back to the server, so that the server can subsequently fail and I can get rid of the TimeoutError. Suggestions appreciated!